src/main/kotlin/org/rust/cargo/toolchain/CargoMessage.kt

```diff
@@ -12,6 +12,9 @@ data class CargoTopMessage(
 ) {
     companion object {
         fun fromJson(json: JsonObject): CargoTopMessage? {
+            if (json.getAsJsonPrimitive("reason") == null) {
+                error("Maformed JSON from Cargo: $json")
```

Comment:
Yeah, I don't expect this to land, I just want to debug what's goining on, because the failure fails to reproduce locally :( 

Hope the test will fail :)